### PR TITLE
PSEC-1599 Adding Route53 Query Logging

### PR DESCRIPTION
**Pull request by ulaskan**
- Adding the route53 logging creation module
- added Aws Create Route53 Public Zones Logs Task
- Creating the route53 create function
- query log arn fix and clean up
- refactored create query log method
- formatting
- added AwsAuditRoute53QueryLogsTask
- end of day work in progress
- fixed circular referencing issue for ComplianceAction
- debugging the enable query log group logic
- end of day work in progress
- enforcement of correct log group refactored
- added retention and tagging and removed redundant files
- Code refactoring to match VPC logging pattern
- fixed logs_boto_client tests
- fixed test_get_route53_client
- fixed hostedzones client unit tests
- moved the test file in correct place and fixed error
- added unit test for enforcement_actions at route53 client
- 100% cova=erage for Route53 client
- added 100 % coverage for AwsClientFactory
- adding the scanner test config unittests
- added unit test to get 100% coverage for aws_compliance_action.py
- Added 53 route 53 enforcement actions test
- added missing unit tests
- Code tidy-up
- Code tidy-up, removed unused mock objects

### CODE REVIEW

File: src/aws_task_runner.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 from src.clients.aws_organizations_client import AwsOrganizationsClient
 from src.clients.aws_ssm_client import AwsSSMClient
 from src.clients.aws_s3_client import AwsS3Client
+from src.clients.aws_hostedZones_client import AwsHostedZonesClient
```
Comment (martibutler) [inline src/aws_task_runner.py:18]: To stay consistent, this should be broken into `aws_hosted_zones_client`.

File: src/clients/aws_client_factory.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
 from src.clients.aws_organizations_client import AwsOrganizationsClient
 from src.clients.aws_ssm_client import AwsSSMClient
 from src.clients.aws_s3_client import AwsS3Client
+from src.clients.aws_hostedZones_client import AwsHostedZonesClient
```
Comment (martibutler) [inline src/clients/aws_client_factory.py:21]: To stay consistent, this should be broken into aws_hosted_zones_client.

File: src/clients/aws_client_factory.py [hunk 3]
```diff
@@ -100,17 +101,26 @@ def get_ec2_client(self, account: Account, role: Optional[str] = None) -> AwsEC2
     def get_route53_boto_client(self, account: Account, role: str) -> BaseClient:
         return self._get_client("route53", account, role)
 
-    def get_route53_client(self, account: Account, role: Optional[str] = None) -> AwsRoute53Client:
-        return AwsRoute53Client(self.get_route53_boto_client(account, role or self._config.route53_role()))
+    def get_route53_client(self, account: Account) -> AwsRoute53Client:
+        return AwsRoute53Client(
+            boto_route53=self.get_hosted_zones_client(account),
+            iam=self.get_iam_client(account),
+            logs=self.get_logs_client(account, region="us-east-1"),
```
Comment (martibutler) [inline src/clients/aws_client_factory.py:108]: Is it always this `us-east-1` region for route53? Could we maybe make this into a constant so it describes exactly why it is hard coded?

Comment (aminebz949904) [reply src/clients/aws_client_factory.py:108]: yes, it's always us-east-1 region for route53, and that is why it makes more sense to hardcode it..

File: src/aws_task_runner.py (hunk 1 above)
Comment (aminebz949904) [reply src/aws_task_runner.py]: renamed

File: src/clients/aws_client_factory.py (hunk 2 above)
Comment (aminebz949904) [reply src/clients/aws_client_factory.py]: renamed

File: src/clients/aws_hosted_zones_client.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            or hosted_zone_query_log["QueryLoggingConfigs"][0]["CloudWatchLogsLogGroupArn"] == ""
+        ):
+            return self._route53.create_query_logging_config(
+                HostedZoneId=hosted_zone_id, CloudWatchLogsLogGroupArn=cloudwatch_logs_loggrouparn
+            )
+
+    def delete_query_logging_config(self, hosted_zone_id: str) -> Any:
+
+        queryLoggingConfig = self._route53.list_query_logging_configs(HostedZoneId=hosted_zone_id)
+        if len(queryLoggingConfig["QueryLoggingConfigs"]) > 0:
```
Comment (cob16) [inline src/clients/aws_hosted_zones_client.py:54]: Will this not delete other query log configs that are unrelated to this code? That could cause problems for infra

File: src/clients/composite/aws_route53_client.py [hunk 5]
```diff
@@ -1,28 +1,110 @@
...
+    def _delete_misconfigured_query_log_actions(
+        self, account: Account, hostedZone: route53Type.Route53Zone
+    ) -> Sequence[ComplianceAction]:
+
+        query_log_arn = (
+            "arn:aws:logs:us-east-1:" + account.identifier + ":log-group:" + self._config.logs_route53_log_group_name()
+        )
+        queryLogActionList = []
+
+        if hostedZone.queryLog != query_log_arn:
```
Comment (cob16) [inline src/clients/composite/aws_route53_client.py:74]: same question

File: src/clients/aws_hosted_zones_client.py (hunk 4 above)
Comment (cob16) [reply src/clients/aws_hosted_zones_client.py:54]: the vpc flow logs only delete quey configs that had our expected name, in that way if left existing config as is (and therefore did not create state drift)

Comment (aminebz949904) [reply src/clients/aws_hosted_zones_client.py:54]: it will delete the query log config related to the specific hosted zone id passed to the method..does this help?

**Review by cob16 — APPROVED** [review]
